src/test/java/fi/helsinki/ubipositioning/trilateration/MeasurementFilterTest.java

```diff
@@ -0,0 +1,73 @@
+package fi.helsinki.ubipositioning.trilateration;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tests for Kalman filter.
+ */
+public class MeasurementFilterTest {
+    @Test
+    public void smooth() {
+        IMeasurementFilter filter = new MeasurementFilter();
+        List<Double> obs = new ArrayList<>();
+
+        for (int i = 0; i < 20; i++) {
+            obs.add((Math.random() * -10) - 20);
```

Comment:
Could we replace Math.random() with int/float literals? I'm not sure it's good idea to use Math.random in unit test inputs, even if it doesn't affect the test results.